back/app/controllers/web_api/v1/user_comments_controller.rb

```diff
@@ -9,22 +9,17 @@ def index
     comment_allowed_ideas = policy_scope(Comment, policy_scope_class: IdeaCommentPolicy::Scope)
       .published
       .where(author_id: params[:user_id])
-    comment_allowed_initiatives = policy_scope(Comment, policy_scope_class: InitiativeCommentPolicy::Scope)
-      .published
-      .where(author_id: params[:user_id])
 
     # Apply pagination to the posts, using the union_posts
     # view and ordering by publication date.
     joined_posts = UnionPost.joins('INNER JOIN comments ON comments.post_id = union_posts.id')
     paged_posts = joined_posts.where(comments: { id: comment_allowed_ideas })
-      .or(joined_posts.where(comments: { id: comment_allowed_initiatives }))
       .order(published_at: :desc)
       .group('union_posts.id, union_posts.published_at') # Remove union_post duplicates
       .select('union_posts.id')
     paged_posts = paginate paged_posts
 
-    # Get the comments, grouped by the corresponding posts
-    # page.
+    # Get the comments, grouped by the corresponding posts page.
```

Comment:
TODO: post_id and post_type -> idea or input

Reply:
Done in Part 3: https://github.com/CitizenLabDotCo/citizenlab/pull/9906